### PR TITLE
[AIRFLOW-6719] Introduce pyupgrade to enforce latest syntax

### PR DESCRIPTION
**Pull request by mik-laj**
Some people love the latest syntax, so I would like to introduce it automatically for the project. These are mainly fstring changes, but not only.

---
Issue link: [AIRFLOW-6719](https://issues.apache.org/jira/browse/AIRFLOW-6719)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by potiuk — COMMENTED** [review]
Love it !

**Comment by mik-laj** [comment]
The first build is unlikely to be valid because pyupgrade introduces changes incompatible with flake8, but I want to check it on CI.

File: airflow/providers/amazon/aws/operators/s3_delete_objects.py [hunk 1]
```diff
@@ -74,4 +74,16 @@ def __init__(
...
+
+        response = s3_hook.delete_objects(bucket=self.bucket, keys=self.keys)
+
+        deleted_keys = [x['Key'] for x in response.get("Deleted", [])]
+        self.log.info("Deleted: %s", deleted_keys)
+
+        if "Errors" in response:
+            errors_keys = [x['Key'] for x in response.get("Errors", [])]
+            raise AirflowException(f"Errors when deleting: {errors_keys}")
+>>>>>>> b18ba328b... [AIRFLOW-6719] Introduce pyupgrade to enforce latest syntax
```
Comment (zhongjiajie) [inline airflow/providers/amazon/aws/operators/s3_delete_objects.py:89]: It fail as you said, not that easy, but this is rebase mistake

File: tests/providers/amazon/aws/operators/test_sftp_to_s3.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def test_sftp_to_s3_operation(self):
         create_file_task = SSHOperator(
             task_id="test_create_file",
             ssh_hook=self.hook,
-            command="echo '{0}' > {1}".format(test_remote_file_content,
+            command="echo '{}' > {}".format(test_remote_file_content,
                                               self.sftp_path),
```
Comment (zhongjiajie) [inline tests/providers/amazon/aws/operators/test_sftp_to_s3.py:85]:
```suggestion
                                            self.sftp_path),
```
```log
tests/providers/amazon/aws/operators/test_sftp_to_s3.py:85:47: E127 continuation line over-indented for visual indent
tests/models/test_taskinstance.py:318:45: E128 continuation line under-indented for visual indent
airflow/ti_deps/deps/runnable_exec_date_dep.py:38:47: E127 continuation line over-indented for visual indent
tests/providers/sftp/operators/test_sftp.py:214:47: E127 continuation line over-indented for visual indent
tests/providers/sftp/operators/test_sftp.py:252:47: E127 continuation line over-indented for visual indent
tests/providers/sftp/operators/test_sftp.py:291:47: E127 continuation line over-indented for visual indent
tests/providers/sftp/operators/test_sftp.py:327:47: E127 continuation line over-indented for visual indent
airflow/cli/commands/dag_command.py:85:50: E127 continuation line over-indented for visual indent
airflow/ti_deps/deps/dag_ti_slots_available_dep.py:33:57: E127 continuation line over-indented for visual indent
airflow/providers/sftp/operators/sftp.py:139:57: E127 continuation line over-indented for visual indent
airflow/providers/sftp/operators/sftp.py:150:57: E127 continuation line over-indented for visual indent
airflow/ti_deps/deps/not_in_retry_period_dep.py:52:41: E127 continuation line over-indented for visual indent
```

**Review by zhongjiajie — COMMENTED** [review]
I think is ok, for now I find out Travis sad due to some nit

**Comment by mik-laj** [comment]
@zhongjiajie I don't have time to finish this, because I've taken care of the scheduler performance. You want to take over?

**Comment by zhongjiajie** [comment]
@mik-laj Sure, but how could I do? submit a new PR to this repo? or create a PR to PolideaInternal:AIRFLOW-6719-pyupgrade?

**Comment by mik-laj** [comment]
@mik-laj  please create new PR to apache/airflow but reuse JIRA number.

**Comment by ashb** [comment]
Closed in favour of #7573

**Pull request closed without merging**